File: generateall.py
"""
 The MIT License (MIT)

Copyright © 2020 Batuhan Gonenc

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the “Software”), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

"""

secki1 = "1234567890-=!@#$%^&*()_+qwertyuiop[]QWERTYUIOP{}asdfghjkl'\ASDFGHJKL|zxcvbnm<>?/~ZXCVBNM"
secki2 = "1234567890-=!@#$%^&*()_+qwertyuiop[]{}asdfghjkl'\|zxcvbnm<>?/~"
secki3 = "1234567890-=!@#$%^&*()_+[]QWERTYUIOP{}'\ASDFGHJKL|<>?/~ZXCVBNM"
secki4 = "1234567890qwertyuiopQWERTYUIOPasdfghjklASDFGHJKLzxcvbnmZXCVBNM"
secki5 = "-=!@#$%^&*()_+[]{}'\|<>?/~"
from random import randint as rInt
from datetime import datetime
import sys


print("---------------------------------------------\nGenerateAll Copyright (c) 2020 Batuhan Gonenc\n---------------------------------------------")
while True:
    i1=input('----------------------------\nchoose one(for options "o"):')
    if i1 =='o':
        print("----------\n1 - generate with all characters\n2 - generate except uppercase\n3 - generate except lowercase \n4 - generate with only alphabet and numbers\n5 - generate except alphabet and numbers\nq - quit\n-----------")
    
    elif i1 == 'q':
        print('good bye...')
        sys.exit()



    elif (i1 == "1"):
        passwords = ""
        secki = secki1
        while 1:
            try:    
                timesinp = input("-----\nhow many passwords do you want? :")
                timesinp = int(timesinp)
                
                break
            except:
                print("-----\nthis input must be a positive integer .")
                continue

            if timesinp <= 0:
                print("-----\nthis input must be a positive integer .")
                continue

        i2 = input("how many characters will your password contain?")
        
        backupi2 = 0
        try:
            i2 = int(i2)
            backupi2 = i2
        except:
            print("#this input must be integer.#")
            continue


        if i2  > 1000:
            print("-----\nwhy did you do this..just..\n-----\ni didn't understand\n-----\ntry something less.\ndon't be a paranoid.")
            continue
        while timesinp > 0:
            i2 = backupi2
            newpassword = ""
            while i2 > 0:
                r_int = rInt(1,len(secki))-1
                new_char = secki[r_int]
                newpassword = newpassword + new_char
                i2 = i2 - 1
            if timesinp == 1:
                print("----------\nyour new passwords are above.")
                break

            print("{}\n".format(newpassword))
            passwords += "{}\n\n".format(newpassword)
            timesinp -= 1

        while 1:
            savein = input("-----\n\ndo you want to save it ? ( y / n ) :")

            if savein == "y":
                filename = input("write a file name:")

                try:
                    file = open("{}.txt".format(filename),"w")
                    
                    now = datetime.now()
                    now = now.strftime("%x %X")
                    content = "{}\n\n\n{} passwords above\n\n{}".format(passwords,backupi2,now)

                    file.write(content)
                    file.close()

                    print("-----\npasswords saved successfully.")

                    break
                except:
                    print("-----\nyou should enter a valid file name.")
                    continue


            elif savein == "n":
                break

            else:
                print("-----\n#unvalid input#")
                continue

    elif (i1 == "2"):
        passwords = ""
        secki = secki2
        while 1:
            try:    
                timesinp = input("-----\nhow many passwords do you want? :")
                timesinp = int(timesinp)
                
                break
            except:
                print("-----\nthis input must be a positive integer .")
                continue

            if timesinp <= 0:
                print("-----\nthis input must be a positive integer .")
                continue

        i2 = input("how many characters will your password contain?")
        
        backupi2 = 0
        try:
            i2 = int(i2)
            backupi2 = i2
        except:
            print("#this input must be integer.#")
            continue


        if i2  > 1000:
            print("-----\nwhy did you do this..just..\n-----\ni didn't understand\n-----\ntry something less.\ndon't be a paranoid.")
            continue
        while timesinp > 0:
            i2 = backupi2
            newpassword = ""
            while i2 > 0:
                r_int = rInt(1,len(secki))-1
                new_char = secki[r_int]
                newpassword = newpassword + new_char
                i2 = i2 - 1
            if timesinp == 1:
                print("----------\nyour new passwords are above.")
                break

            print("{}\n".format(newpassword))
            passwords += "{}\n\n".format(newpassword)
            timesinp -= 1

        while 1:
            savein = input("-----\n\ndo you want to save it ? ( y / n ) :")

            if savein == "y":
                filename = input("write a file name:")

                try:
                    file = open("{}.txt".format(filename),"w")
                    
                    now = datetime.now()
                    now = now.strftime("%x %X")
                    content = "{}\n\n\n{} passwords above\n\n{}".format(passwords,backupi2,now)

                    file.write(content)
                    file.close()

                    print("-----\npasswords saved successfully.")

                    break
                except:
                    print("-----\nyou should enter a valid file name.")
                    continue


            elif savein == "n":
                break

            else:
                print("-----\n#unvalid input#")
                continue
    elif (i1 == "3"):
        passwords = ""
        secki = secki3
        while 1:
            try:    
                timesinp = input("-----\nhow many passwords do you want? :")
                timesinp = int(timesinp)
                
                break
            except:
                print("-----\nthis input must be a positive integer .")
                continue

            if timesinp <= 0:
                print("-----\nthis input must be a positive integer .")
                continue

        i2 = input("how many characters will your password contain?")
        
        backupi2 = 0
        try:
            i2 = int(i2)
            backupi2 = i2
        except:
            print("#this input must be integer.#")
            continue


        if i2  > 1000:
            print("-----\nwhy did you do this..just..\n-----\ni didn't understand\n-----\ntry something less.\ndon't be a paranoid.")
            continue
        while timesinp > 0:
            i2 = backupi2
            newpassword = ""
            while i2 > 0:
                r_int = rInt(1,len(secki))-1
                new_char = secki[r_int]
                newpassword = newpassword + new_char
                i2 = i2 - 1
            if timesinp == 1:
                print("----------\nyour new passwords are above.")
                break

            print("{}\n".format(newpassword))
            passwords += "{}\n\n".format(newpassword)
            timesinp -= 1

        while 1:
            savein = input("-----\n\ndo you want to save it ? ( y / n ) :")

            if savein == "y":
                filename = input("write a file name:")

                try:
                    file = open("{}.txt".format(filename),"w")
                    
                    now = datetime.now()
                    now = now.strftime("%x %X")
                    content = "{}\n\n\n{} passwords above\n\n{}".format(passwords,backupi2,now)

                    file.write(content)
                    file.close()

                    print("-----\npasswords saved successfully.")

                    break
                except:
                    print("-----\nyou should enter a valid file name.")
                    continue


            elif savein == "n":
                break

            else:
                print("-----\n#unvalid input#")
                continue
    elif (i1 == "4"):
        passwords = ""
        secki = secki4
        while 1:
            try:    
                timesinp = input("-----\nhow many passwords do you want? :")
                timesinp = int(timesinp)
                
                break
            except:
                print("-----\nthis input must be a positive integer .")
                continue

            if timesinp <= 0:
                print("-----\nthis input must be a positive integer .")
                continue

        i2 = input("how many characters will your password contain?")
        
        backupi2 = 0
        try:
            i2 = int(i2)
            backupi2 = i2
        except:
            print("#this input must be integer.#")
            continue


        if i2  > 1000:
            print("-----\nwhy did you do this..just..\n-----\ni didn't understand\n-----\ntry something less.\ndon't be a paranoid.")
            continue
        while timesinp > 0:
            i2 = backupi2
            newpassword = ""
            while i2 > 0:
                r_int = rInt(1,len(secki))-1
                new_char = secki[r_int]
                newpassword = newpassword + new_char
                i2 = i2 - 1
            if timesinp == 1:
                print("----------\nyour new passwords are above.")
                break

            print("{}\n".format(newpassword))
            passwords += "{}\n\n".format(newpassword)
            timesinp -= 1

        while 1:
            savein = input("-----\n\ndo you want to save it ? ( y / n ) :")

            if savein == "y":
                filename = input("write a file name:")

                try:
                    file = open("{}.txt".format(filename),"w")
                    
                    now = datetime.now()
                    now = now.strftime("%x %X")
                    content = "{}\n\n\n{} passwords above\n\n{}".format(passwords,backupi2,now)

                    file.write(content)
                    file.close()

                    print("-----\npasswords saved successfully.")

                    break
                except:
                    print("-----\nyou should enter a valid file name.")
                    continue


            elif savein == "n":
                break

            else:
                print("-----\n#unvalid input#")
                continue
    elif (i1 == "5"):
        passwords = ""
        secki = secki5
        while 1:
            try:    
                timesinp = input("-----\nhow many passwords do you want? :")
                timesinp = int(timesinp)
                
                break
            except:
                print("-----\nthis input must be a positive integer .")
                continue

            if timesinp <= 0:
                print("-----\nthis input must be a positive integer .")
                continue

        i2 = input("how many characters will your password contain?")
        
        backupi2 = 0
        try:
            i2 = int(i2)
            backupi2 = i2
        except:
            print("#this input must be integer.#")
            continue


        if i2  > 1000:
            print("-----\nwhy did you do this..just..\n-----\ni didn't understand\n-----\ntry something less.\ndon't be a paranoid.")
            continue
        while timesinp > 0:
            i2 = backupi2
            newpassword = ""
            while i2 > 0:
                r_int = rInt(1,len(secki))-1
                new_char = secki[r_int]
                newpassword = newpassword + new_char
                i2 = i2 - 1
            if timesinp == 1:
                print("----------\nyour new passwords are above.")
                break

            print("{}\n".format(newpassword))
            passwords += "{}\n\n".format(newpassword)
            timesinp -= 1

        while 1:
            savein = input("-----\n\ndo you want to save it ? ( y / n ) :")

            if savein == "y":
                filename = input("write a file name:")

                try:
                    file = open("{}.txt".format(filename),"w")
                    
                    now = datetime.now()
                    now = now.strftime("%x %X")
                    content = "{}\n\n\n{} passwords above\n\n{}".format(passwords,backupi2,now)

                    file.write(content)
                    file.close()


                    print("-----\npasswords saved successfully.")
                    break
                except:
                    print("-----\nyou should enter a valid file name.")
                    continue


            elif savein == "n":
                break

            else:
                print("-----\n#unvalid input#")
                continue

  
    else:
        print("#unvalid input#")

File: generateall.cpp
#include <iostream>
#include <stdlib.h>
#include <time.h>
#include <fstream>


using namespace std;


//rating algorithm

double rater(string pw)
{
	int SIZE = pw.size();
	const int pw_size{SIZE};


    string basics{"1234567890qwertyuiopQWERTYUIOPasdfghjklASDFGHJKLzxcvbnmZXCVBNM"};

    string specials{ "&$'|=!@#%^*()+[]{}?/:;<>~" } ;


    double basicnum{0};
    double specnum{0};
    double rating {0};


    for(int i{0} ; i < 62 ; i++)
    {

    	for(int j{0} ; j < pw_size ; j++)
    	{

    		if(basics[i] == pw[j]){
    			basicnum++;
				}
    	}
    }


    for(int i{0} ; i < 26 ; i++)
    {

    	for(int j{0} ; j < pw_size ; j++)
    	{

    		if(specials[i] == pw[j]){
    			specnum++;
				}
    	}
    }


	if(specnum == 0)
	specnum++;


    rating = basicnum / specnum ;

    return rating;}


int main()
{
	//making the rand func ready
	srand(time(NULL));

	int SIZE;

	string secki1 = "1234567890-=!@#$%^&*()_+qwertyuiop[]QWERTYUIOP{}asdfghjkl'ASDFGHJKL|zxcvbnm<>?/~ZXCVBNM";
	SIZE = secki1.size() ;
	const int size1{SIZE};


	string secki2 = "1234567890-=!@#$%^&*()_+qwertyuiop[]{}asdfghjkl'|zxcvbnm<>?/~";
	SIZE = secki2.size() ;
	const int size2{SIZE};

	string secki3 = "1234567890-=!@#$%^&*()_+[]QWERTYUIOP{}'ASDFGHJKL|<>?/~ZXCVBNM";
	SIZE = secki3.size() ;
	const int size3{SIZE};

	string secki4 = "1234567890qwertyuiopQWERTYUIOPasdfghjklASDFGHJKLzxcvbnmZXCVBNM";
	SIZE = secki4.size() ;
	const int size4{SIZE};

	string secki5 =  "&$'|=!@#%^*()+[]{}?/:;<>~";
	SIZE = secki5.size() ;
	const int size5{SIZE};


	int ptimes{0}, ctimes{0}, tempctimes{0}, select{0}, randint{0};



	cout <<"GenerateAll\n\n0 - quit\n1 - all characters\n2 - except uppercase\n3 - except lowercase\n4 - only alphabet\n5 - only special\n----------\n";

	for(;;)
	{

		cout << "\n\nchoose :" ; cin >> select ;

		if(select == 0){
			break;
		}


		else if(select == 1)
		{	
			string filename;
			int flag{0};
			
			cout << "do you want to save it ? ( 1 / 0 ) :" ; cin >> flag ;
			
			if(flag == 1){
				cout << "give a file name : ";cin >> filename;
				}

			cout << "how many passwords do you want ? :" ; cin >> ptimes ;

			cout << "how many chars they will contain ? :" ; cin >> ctimes ;
			cout << endl << endl ;


			if(ptimes == 0)
				ptimes++;

			if(ctimes < 4)
				ctimes += 8;

			char password[ctimes]{'a'} ;
			
			
			//file things
			
			ofstream fout;
			fout.open(filename);

			for(int i{ptimes} ; i > 0; i--)
			{

				tempctimes = ctimes ;

				for(int j{tempctimes} ; j > 0 ; j--)
				{


					randint = (rand() % size1) - 1;

					password[ ctimes - j ] = secki1[randint] ;

				}

				string line{""};
				string ratepassword{""};

				for(int k{0} ; k < ctimes ; k++){
					cout << password[k] ;
					ratepassword += password[k];
					line += password[k];
							}

				cout << " ---> rating : no data\n" ;
				
				line += " ---> rating : no data\n" ;
				
				fout << line ;

			}
			fout.close();
			if(flag == 1)
			cout << "\n---\ncheck the file.\nall passwords saved.\n" ;


		}




		else if(select == 2)
		{
			string filename;
			int flag{0};
			
			cout << "do you want to save it ? ( 1 / 0 ) :" ; cin >> flag ;
			
			if(flag == 1){
				cout << "give a file name : ";cin >> filename;
				}	

			cout << "how many passwords do you want ? :" ; cin >> ptimes ;

			cout << "how many chars they will contain ? :" ; cin >> ctimes ;
			cout << endl << endl ;

			if(ptimes == 0)
				ptimes++;

			if(ctimes < 4)
				ctimes += 8;

			char password[ctimes]{'a'} ;
			
			
			//file things
			
			ofstream fout;
			fout.open(filename);

			for(int i{ptimes} ; i > 0; i--)
			{

				tempctimes = ctimes ;

				for(int j{tempctimes} ; j > 0 ; j--)
				{


					randint = (rand() % size2) - 1;

					password[ ctimes - j ] = secki2[randint] ;

				}
				string line{""};
				string ratepassword{""};

				for(int k{0} ; k < ctimes ; k++){
					cout << password[k] ;
					ratepassword += password[k];
					line += password[k];
							}

				cout << " ---> rating : no data\n" ;
				
				line += " ---> rating : no data\n" ;
				
				fout << line ;

			}
			fout.close();
			if(flag == 1)
			cout << "\n---\ncheck the file.\nall passwords saved.\n" ;


		}



		else if(select == 3)
		{
			string filename;
			int flag{0};
			
			cout << "do you want to save it ? ( 1 / 0 ) :" ; cin >> flag ;
			
			if(flag == 1){
				cout << "give a file name : ";cin >> filename;
				}
			cout << "how many passwords do you want ? :" ; cin >> ptimes ;

			cout << "how many chars they will contain ? :" ; cin >> ctimes ;
			cout << endl << endl ;

			if(ptimes == 0)
				ptimes++;

			if(ctimes < 4)
				ctimes += 8;

			char password[ctimes]{'a'} ;
			
			
			//file things
			
			ofstream fout;
			fout.open(filename);

			for(int i{ptimes} ; i > 0; i--)
			{

				tempctimes = ctimes ;

				for(int j{tempctimes} ; j > 0 ; j--)
				{


					randint = (rand() % size3) - 1;

					password[ ctimes - j ] = secki3[randint] ;

				}
				string line{""};
				string ratepassword{""};

				for(int k{0} ; k < ctimes ; k++){
					cout << password[k] ;
					ratepassword += password[k];
					line += password[k];
							}

				cout << " ---> rating : no data\n" ;
				
				line += " ---> rating : no data\n" ;
				
				fout << line ;

			}
			fout.close();
			if(flag == 1)
			cout << "\n---\ncheck the file.\nall passwords saved.\n" ;


		}



		else if(select == 4)
		{
			string filename;
			int flag{0};
			
			cout << "do you want to save it ? ( 1 / 0 ) :" ; cin >> flag ;
			
			if(flag == 1){
				cout << "give a file name : ";cin >> filename;
				}
			cout << "how many passwords do you want ? :" ; cin >> ptimes ;

			cout << "how many chars they will contain ? :" ; cin >> ctimes ;
			cout << endl << endl ;

			if(ptimes == 0)
				ptimes++;

			if(ctimes < 4)
				ctimes += 8;
				
				
			//file things
			
			ofstream fout;
			fout.open(filename);

			char password[ctimes]{'a'} ;

			for(int i{ptimes} ; i > 0; i--)
			{

				tempctimes = ctimes ;

				for(int j{tempctimes} ; j > 0 ; j--)
				{


					randint = (rand() % size4) - 1;

					password[ ctimes - j ] = secki4[randint] ;

				}

	string line{""};
				string ratepassword{""};

				for(int k{0} ; k < ctimes ; k++){
					cout << password[k] ;
					ratepassword += password[k];
					line += password[k];
							}

				cout << " ---> rating : no data\n" ;
				
				line += " ---> rating : no data\n" ;
				
				fout << line ;

			}
			
			fout.close();
			if(flag == 1)
			cout << "\n---\ncheck the file.\nall passwords saved.\n" ;


		}



		else if(select == 5)
		{
			string filename;
			int flag{0};
			
			cout << "do you want to save it ? ( 1 / 0 ) :" ; cin >> flag ;
			
			if(flag == 1){
				cout << "give a file name : ";cin >> filename;
				}
				
			cout << "how many passwords do you want ? :" ; cin >> ptimes ;

			cout << "how many chars they will contain ? :" ; cin >> ctimes ;
			cout << endl << endl ;

			if(ptimes == 0)
				ptimes++;

			if(ctimes < 4)
				ctimes += 8;

			char password[ctimes]{'a'} ;


			//file things
			
			ofstream fout;
			fout.open(filename);
			
			for(int i{ptimes} ; i > 0; i--)
			{

				tempctimes = ctimes ;

				for(int j{tempctimes} ; j > 0 ; j--)
				{


					randint = (rand() % size5) - 1;

					password[ ctimes - j ] = secki5[randint] ;

				}
				string line{""};
				string ratepassword{""};

				for(int k{0} ; k < ctimes ; k++){
					cout << password[k] ;
					ratepassword += password[k];
					line += password[k];
							}

				cout << " ---> rating : no data\n" ;
				
				line += " ---> rating : no data\n" ;
				
				fout << line ;

			}
			
			fout.close();
			if(flag == 1)
			cout << "\n---\ncheck the file.\nall passwords saved.\n" ;


		}




	}


	return 0 ;
}
